### ISSUE TITLE
Choosing a different folder to read epicea produces a DNU

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#17521]
- Sources
- Click on Directory
- Boum
- 
![Capture 2024-12-14 at 17 30 31](https://github.com/user-attachments/assets/125b298d-545a-47d7-8b92-0632f3068692)

**Comment by Ducasse** [comment]
```
FileLocator(Object)>>doesNotUnderstand: #localizedForPresenter:
FileLocator>>doesNotUnderstand: #localizedForPresenter:
[ :w | 
		w setLabel: (aString ifNotNil: [ :theString | 
				 theString localizedForPresenter: self presenter ]) ] in SpMorphicModalWindowAdapter(SpMorphicWindowAdapter)>>title: in Block: [ :w | ...
FullBlockClosure(BlockClosure)>>cull:
SpDialogWindowMorph(ProtoObject)>>ifNotNil:
SpMorphicModalWindowAdapter(SpAbstractAdapter)>>widgetDo:
SpMorphicModalWindowAdapter(SpMorphicWindowAdapter)>>title:
[ :anAdapter | anAdapter title: self title ] in SpModalWindowPresenter(SpWindowPresenter)>>updateTitle in Block: [ :anAdapter | anAdapter title: self title ]
SpModalWindowPresenter(SpAbstractPresenter)>>withAdapterDo:
SpModalWindowPresenter(SpWindowPresenter)>>updateTitle
[ 
			self allPresenters do: [ :each | each announceDisplayed ].
			self updateTitle ] in [ :anAdapter | 
		anAdapter openWithDeferredAction: [ 
			self allPresenters do: [ :each | each announceDisplayed ].
			self updateTitle ] ] in SpModalWindowPresenter(SpWindowPresenter)>>openWithLayout: in Block: [ ...
SpMorphicModalWindowAdapter>>openWithDeferredAction:
[ :anAdapter | 
		anAdapter openWithDeferredAction: [ 
			self allPresenters do: [ :each | each announceDisplayed ].
			self updateTitle ] ] in SpModalWindowPresenter(SpWindowPresenter)>>openWithLayout: in Block: [ :anAdapter | ...
SpModalWindowPresenter(SpAbstractPresenter)>>withAdapterDo:
SpModalWindowPresenter(SpWindowPresenter)>>openWithLayout:
SpModalWindowPresenter(SpWindowPresenter)>>openWith:
SpModalWindowPresenter(SpWindowPresenter)>>open
StOpenDirectoryPresenter(SpAbstractPresenter)>>openModal
StOpenDirectoryPresenter(StFileDialogPresenter)>>openModal
SpMorphicBackend>>executeOpenDirectoryDialog:
SpMorphicBackend>>openFileDialog:
SpFileDialog>>openModal
SpNullApplication(SpApplication)>>selectDirectoryTitle:
EpLogNodeGraphPresenter>>selectDirectory
EpUnifiedBrowserPresenter>>selectDirectory
[ self selectDirectory ] in EpUnifiedBrowserPresenter>>initializeToolbar in Block: [ self selectDirectory ]
FullBlockClosure(BlockClosure)>>cull:
SpMorphicToolbarButtonAdapter>>execute
SpToolbarButtonMorph(PluggableButtonMorph)>>performAction:
[:m |
		(m containsPoint: evt cursorPoint) ifTrue: [m enabled ifTrue: [ m performAction: evt ]]] in SpToolbarButtonMorph(PluggableButtonMorph)>>mouseUp: in Block: [:m |...
Array(SequenceableCollection)>>do:
SpToolbarButtonMorph(PluggableButtonMorph)>>mouseUp:
SpToolbarButtonMorph(Morph)>>handleMouseUp:
MouseButtonEvent>>sentTo:
SpToolbarButtonMorph(Morph)>>handleEvent:
SpToolbarButtonMorph(Morph)>>handleFocusEvent:
[
		result := focusHolder handleFocusEvent: transformedEvent.
	] in HandMorph>>sendFocusEvent:to:clear: in Block: [...
FullBlockClosure(BlockClosure)>>on:do:
WorldMorph>>becomeActiveDuring:
HandMorph>>sendFocusEvent:to:clear:
HandMorph>>sendEvent:focus:clear:
HandMorph>>sendMouseEvent:
HandMorph>>handleEvent:
[
		(morphicWorld activeHand isNotNil and: [ anEvent hand isNotNil ]) ifTrue: [
			morphicWorld activeHand handleEvent: anEvent
		]
	] in OSWindowMorphicEventHandler>>dispatchMorphicEvent: in Block: [...
WorldState>>runStepMethodsIn:
WorldMorph>>runStepMethods
WorldState>>doOneCycleFor:
WorldMorph>>doOneCycleNow
WorldMorph>>doOneCycle
[
		| extraWorldsToDraw |
		extraWorldsToDraw := ExtraWorldListMutex critical: [
			                     self extraWorldList ].
		extraWorldsToDraw do: [ :world | world doOneCycle ].

		(self currentWorld isNotNil and: [
			 (extraWorldsToDraw includes: self currentWorld) not ]) ifTrue: [
			self currentWorld doOneCycle ] ] in WorldMorph class>>doOneCycle in Block: [...
FullBlockClosure(BlockClosure)>>ensure:
WorldState class>>doDrawCycleWith:
WorldMorph class>>doOneCycle
MorphicRenderLoop>>doOneCycle
MorphicRenderLoop>>doOneCycleWhile:
[ MorphicRenderLoop new doOneCycleWhile: [ true ] ] in MorphicUIManager>>spawnNewProcess in Block: [ MorphicRenderLoop new doOneCycleWhile: [ tru[..]
[self value.
			"IMPORTANT: Do not step over next line of code. See method comments for details"
			Processor terminateRealActive] in FullBlockClosure(BlockClosure)>>newProcess in Block: [self value....

```

**Comment by Ducasse** [comment]
I spent time and I have a fix but I will not give it because I cannot write a test. 
And since there are not tests on openModal (not a single one) I think that openModal was never tested.  
So I will not contribute to Pharo under such condition.
Sorry guys I spent time finding the bug, fixing it, then trying to write a test and I could not and I realize
that we have a clear mismatch in values in software development. So I stop.
